src/main/java/edu/kit/datamanager/metastore2/configuration/WebSecurityConfig.java

```diff
@@ -67,18 +68,17 @@ public class WebSecurityConfig {
   public WebSecurityConfig() {
   }
 
-  @Bean
-  public AuthenticationManager authenticationManager(AuthenticationConfiguration authenticationConfiguration) throws Exception {
-    return authenticationConfiguration.getAuthenticationManager();
-  }
-
-  @Bean
-  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-    AuthenticationManager authenticationManager = authenticationManager(http.getSharedObject(AuthenticationConfiguration.class));
-    HttpSecurity httpSecurity = http.authorizeRequests()
-            .antMatchers(HttpMethod.OPTIONS, "/**").
-            permitAll().
-            and().
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    HttpSecurity httpSecurity = http.authorizeRequests().
+            requestMatchers(EndpointRequest.to(
+                    InfoEndpoint.class,
+                    HealthEndpoint.class
+            )).permitAll().
```

Comment:
I might be wrong, but it reads like you give permissions to everyone for the health+info endpoints and then restrict it to ROLE_ACTUATOR (and role_admin) for all other endpoints. Shouldn't it be the other way around?

Reply:
I think the reason is, that it's a filter chain which is left as soon as one rule matches. Currently, I suppose that everybody can access info and health, health details only when authenticated, and all other endpoints of having one of the special roles. Seems fine to me.

Reply:
The first point is only visible in application.properties. in principle the order should be as follows:

* everybody sees info and basis health
* authenticated users see in addition complete health
* special roles also see other actuators, e.g., beans, loggers, configuration properties

Does this make sense?

Reply:
It does. I also found the management settings in the config file, thanks. It's a bit confusing it is not in the same place, but I see the idea behind it. There is still one point of confusion. Right below this line, it is like this:

```java
.requestMatchers(EndpointRequest.toAnyEndpoint())
.hasAnyRole("ADMIN", "ACTUATOR")
```

But wouldn't this also affect the actual application endpoints / APIs? Is being admin/actuator required to use the metastore API, and why do actuator roles need access?